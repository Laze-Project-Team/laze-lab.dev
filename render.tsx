import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
