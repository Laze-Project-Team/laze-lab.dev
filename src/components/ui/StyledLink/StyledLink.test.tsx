import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { render } from '@/lib/test/render';

import { StyledLink } from '.';

it('should be rendered correctly', () => {
  const { container } = render(<StyledLink href="/" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('should have "link" role', () => {
  render(<StyledLink href="/" />);
  expect(screen.getByRole('link')).toBeInTheDocument();
});

it('should render children', () => {
  const children = "Hey I'm children";
  render(<StyledLink href="/">{children}</StyledLink>);
  expect(screen.getByText(children)).toBeInTheDocument();
});

it('should be passed href', () => {
  const href = '/ref_to_somewhere';
  render(<StyledLink href={href} />);
  expect(screen.getByRole('link').closest('a')).toHaveAttribute('href', href);
});
