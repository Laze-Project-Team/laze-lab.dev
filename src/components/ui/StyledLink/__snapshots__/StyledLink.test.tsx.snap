// Vitest Snapshot v1

exports[`should be rendered correctly 1`] = `
<a
  class="css-1qn4jyq-PresentialStyledLink-PresentialStyledLink"
  href="/"
/>
`;
